deepchem/models/torch_models/layers.py

```diff
@@ -6164,3 +6164,178 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
+        torch.tensor(eps))
+    cosine_dist = 1 - cos_similarity
+
+    return cosine_dist
+
+
+class AttnLSTMEmbedding(nn.Module):
+    """
+    Implements AttnLSTM as in matching networks paper.
+
+    The AttnLSTM embedding adjusts two sets of vectors, the "test" and
+    "support" sets. The "support" consists of a set of evidence vectors.
+    Think of these as the small training set for low-data machine
+    learning.  The "test" consists of the queries we wish to answer with
+    the small amounts of available data. The AttnLSTMEmbdding allows us to
+    modify the embedding of the "test" set depending on the contents of
+    the "support".  The AttnLSTMEmbedding is thus a type of learnable
+    metric that allows a network to modify its internal notion of
+    distance.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import torch
+    >>> import deepchem.models.torch_models.layers as layers
+    >>> max_depth = 5
+    >>> n_test = 5
+    >>> n_support = 11
+    >>> n_feat = 10
+    >>> test = np.random.rand(n_test, n_feat).astype(np.float32)
+    >>> support = np.random.rand(n_support, n_feat).astype(np.float32)
+    >>> layer = layers.AttnLSTMEmbedding(n_test, n_support, n_feat, max_depth)
+    >>> test_out, support_out = layer([test, support])
+    >>> test_out.shape
+    torch.Size([5, 10])
+    >>> support_out.shape
+    torch.Size([11, 10])
+
+    References
+    ----------
+    .. [1] Vinyals, Oriol, et al. "Matching networks for one shot learning."
+        Advances in neural information processing systems. 2016.
+    .. [2] Vinyals, Oriol, Samy Bengio, and Manjunath Kudlur. "Order matters:
+        Sequence to sequence for sets." arXiv preprint arXiv:1511.06391 (2015).
+   
+    """
+
+    def __init__(self, n_test, n_support, n_feat, max_depth, **kwargs):
+        """
+        Parameters
+        ----------  
+        n_support: int
+            Size of support set.
+        n_test: int
+            Size of test set.
+        n_feat: int
+            Number of features per atom
+        max_depth: int
+            Number of "processing steps" used by sequence-to-sequence for sets model.
+        """
+        super(AttnLSTMEmbedding, self).__init__(**kwargs)
+        self.max_depth = max_depth
+        self.n_test = n_test
+        self.n_support = n_support
+        self.n_feat = n_feat
+
+        n_feat = self.n_feat
+        self.lstm = LSTMStep(n_feat, 2 * n_feat)
+        self.q_init = torch.zeros([self.n_test, n_feat])
+        self.states_init = self.lstm.get_initial_states([self.n_test, n_feat])
+        self.built = True
+
+    def __repr__(self):
+        return (
+            f'{self.__class__.__name__}(n_test={self.n_test}, n_support={self.n_support}, n_feat={self.n_feat}, max_depth={self.max_depth})'
+        )
+
+    def forward(self, inputs):
+        """Execute this layer on input tensors.
+
+        Parameters
+        ----------
+        inputs: list
```

Comment:
@Dragonwagon18 Shorten the description of input parameter, it will cause the rendered description to look cluttered. You've mentioned n_test, feat and support above in the model's init parameter description

Reply:
done